src/pip/_internal/cli/base_command.py

```diff
@@ -68,6 +73,10 @@ def __init__(self, name, summary, isolated=False):
         self.summary = summary
         self.parser = ConfigOptionParser(**parser_kw)
 
+        self.tempdir_registry = (
+            None
+        )  # type: Optional[TempDirectoryTypeRegistry]
```

Comment:
Gosh this style gives me goosebumps every time. Would it be a good idea to add a typedef and make this fit in one line?

Reply:
Not worth it IMO since it would necessarily need to be under an `if MYPY_CHECK_RUNNING` so either an ugly block here or the def for this local variable is put at the top of the file, both would be more confusing to me at first glance.

Reply:
OK

Reply:
Thanks, `TempDirRegistry` ended up looking fine.